builders.ncl

```diff
@@ -81,195 +70,158 @@ let { NickelDerivation, Derivation, .. } = import "contracts.ncl" in
           |> std.record.map
             (
               fun _n xs =>
-                # Intersperse ':' between environment variable fragments
-                let values = std.record.values xs in
-                let first = std.array.first values in
-                values
-                |> std.array.drop_first
-                |> std.array.fold_left (fun acc value => nix-s%"%{acc}:%{value}"%) first
+                # Intersperse ' ' between environment variable fragments
+                concat_strings_sep " " (std.record.values xs)
             ),
-      } | (NickelPkg & { packages | { _ : Derivation } }),
+      } | NickelPkg,
+    },
+
+  Shell
+    | doc m%"
+      A derivation that is to be used as a shell, e.g. with `nix develop`.
+      Analogous to `mkShell`.
+    "%
+    = NixpkgsPkg & {
+      packages | doc "Packages to be added to the shell, setting PATH, LD_LIBRARY_PATH and other variables as needed" = {},
+      hooks | doc "Bash scripts to run when entering the shell" = {},
+
+      output = {
+        name | default = "shell",
+        version | default = "dev",
+
+        env.buildCommand = nix-s%"
+          echo "This derivation is not supposed to be built" 1>&2 1>/dev/null
+          exit 1
+        "%,
+        env.shellHook = concat_strings_sep "\n" (std.record.values hooks),
+        structured_env.buildInputs = packages,
+      } | NickelPkg,
+    },
```

Comment:
There's a bit of confusion here between what's an input and what's an output. But it's already the case atm, so it's more a note for later than a real concern with this PR